python/lsst/ts/linearstage/controllers/zaber_lst.py

```diff
@@ -19,23 +19,229 @@
+        The tcpip client that commands the device.
+    mock_server : `None` | `LinearStageServer`
+        The mock server that's started if in simulation mode.
+    position : `None` | `float`
+        The position of the stage along the axis.
+
+    """
+
+    def __init__(
+        self, config: types.SimpleNamespace, log: logging.Logger, simulation_mode: int
+    ) -> None:
+        super().__init__(config, log, simulation_mode)
+        self.client: Connection | None = None
+        self.mock_server: LinearStageServer | None = None
+        self.position: float | None = None
+        self.device: Device | None = None
+
+    @property
+    def connected(self) -> bool:
+        """Is the client connected?"""
+        # Assume client is connected
+        if self.client is not None:
+            return True
+        else:
+            return False
+
+    async def connect(self) -> None:
+        """Connect to the device."""
+        if self.simulation_mode:
+            self.mock_server = LinearStageServer(port=0, log=self.log)
+            await self.mock_server.start_task
+            self.config.port = self.mock_server.port
+        try:
+            self.client = await Connection.open_tcp_async(
+                host_name=self.config.hostname, port=self.config.port
+            )
+        except ConnectionFailedException:
+            self.log.exception("Failed to connect to host/port.")
+            raise RuntimeError(f"Unable to connect to host {self.config.hostname}.")
+        self.device = self.client.get_device(self.config.daisy_chain_address)
+        self.log.debug(f"{self.device=}")
+        await self.device.identify_async()
+        self.log.debug(f"{self.device=}")
+
+    async def disconnect(self) -> None:
+        """Disconnect from the device."""
+        if self.client is not None:
+            self.client.close()
+        self.client = None
+        self.device = None
+        if self.simulation_mode:
+            if self.mock_server is not None:
+                await self.mock_server.close()
+            self.mock_server = None
+
+    async def move_relative(self, value: float) -> None:
+        """Move device to position with relative target.
+
+        Parameters
+        ----------
+        value : `float`
+            The amount to move by.
+        """
+        if self.device is None:
+            raise RuntimeError("Device has not been received.")
+        for axis_index in range(self.device.axis_count):
+            axis = self.device.get_axis(axis_index + 1)
+            if axis.axis_type != AxisType.UNKNOWN:
+                try:
+                    await axis.move_relative_async(
+                        position=value, unit=Units.LENGTH_MILLIMETRES
+                    )
+                except CommandFailedException:
+                    self.log.exception("Move relative failed.")
+                    raise
+            else:
+                self.log.info(f"Skipping {axis.axis_number=}.")
```

Comment:
I see that this same if-block is repeated several times in the code, with a different function call. Please extract to a function that retrieves the axis and use that so there is no repetitive code.